tests/components/backup/test_store.py

```diff
@@ -98,7 +98,7 @@ def mock_delay_save() -> Generator[None]:
                     }
                 ],
                 "config": {
-                    "agents": {"test.remote": {"protected": True}},
+                    "agents": {"test.remote": {"protected": True, "retention": None}},
```

Comment:
Why is this changed?

Reply:
This data has major version 2 so must already have all minor version migrations. Since we added a minor migration we need to update the existing store to match that.